skimage/restoration/tests/test_unwrap.py

```diff
@@ -109,10 +113,12 @@ def check_wrap_around(ndim, axis):
                         image_unwrap_wrap_around[index_last])
 
 
-def test_wrap_around():
-    for ndim in (2, 3):
-        for axis in range(ndim):
-            yield check_wrap_around, ndim, axis
+dim_axis = [(ndim, axis) for ndim in (2, 3) for axis in range(ndim)]
```

Comment:
This is a pain to read :). Maybe reformulate somehow?

Reply:
Unfortunately, that's just how double comprehensions look in Python :/

Reply:
What about the example below? The logic is as before my changes but moved to a separate "builder" function:
```python
def generate_dim_and_axis():
    for ndim in (2, 3):
        for axis in range(ndim):
            yield ndim, axis
dim_axis = list(generate_dim_and_axis())
```
I personally prefer the double comprehension version, but open to change :-)

Reply:
I prefer the original too.